spec/controllers/campaigns_controller_spec.rb

```diff
@@ -123,4 +123,128 @@
       end
     end
   end
+
+  describe 'PUT #publish' do
```

Comment:
and please review approach, you are testing controller, not model.  The enough will be to test whether campaign received some event or not.

Reply:
Transition specs you can put to the model.

Reply:
Moved and deleted.